pkg/oauth/tokenrequest/challengehandlers/basicauth.go

```diff
@@ -107,6 +111,10 @@ func (c *BasicChallengeHandler) HandleChallenge(requestURL string, headers http.
 			w = os.Stdout
 		}
 
+		if len(c.WebConsoleURL) > 0 {
+			fmt.Fprintf(w, "Console URL: %s/console\n", c.WebConsoleURL)
```

Comment:
In oc, `Host` field is used for printing this message https://github.com/openshift/oc/blob/a65a29a57e4d213df00e2ee1fa3641dc19912e3b/pkg/helpers/tokencmd/basicauth.go#L97 and as far as I can see `Host` is already defined. Is this new field `WebConsoleURL` different than `Host`?

Reply:
the `Host` field is used along with an addition of the `/console` path. The code you refer to supplies the whole ClusterVersionGetter to the challengers, and copy-pastes all the logic.

My proposal is to do the cluster version eval once _outside_ the challengers, and these would only print the console URL if supplied.

As such, I am replacing the clusterVersionGetter with just a string